File: 2022/01/Perl/bin/main.pl
#!perl

use v5.14;
use warnings;
use constant FILENAME => $ENV{ADVENT_INPUT};

my @calories = do {
	local @ARGV = FILENAME;
	my ( $i, @data ) = 0;
	while ( <> ) {
		chomp;
		/[0-9]/ ? ( $data[$i] += $_ ) : ++$i;
	}
	@data;
};

my @sorted =
	sort { $b->[1] <=> $a->[1] }
	map { [ $_, $calories[$_] ] }
	0 .. $#calories;

say "PART1: ", $sorted[0][1];

my $total = 0;
$total += $sorted[$_][1] for 0..2;
say "PART2: ", $total;

File: 2022/01/Javascript/main.js
const fs = require( 'fs' );

fs.readFile( process.env.ADVENT_INPUT, 'utf8', ( err, data ) => {
	if ( err ) {
		console.error( err );
		return;
	}

	var elf_no = 0;
	var calories = [];
	data.split( /\r?\n/ ).forEach( line => {
		if ( line == "" ) {
			elf_no++;
		}
		else {
			if ( ! calories[elf_no] ) {
				calories[elf_no] = 0;
			}
			calories[elf_no] += parseInt( line, 10 );
		}
	} );

	var sorted = calories.map( ( c, e ) => [ e, c ] );
	sorted.sort( ( a, b ) => ( a[1] < b[1] ) ? 1 : ( ( a[1] > b[1] ) ? -1 : 0 ) );

	console.log( `PART1: ${sorted[0][1]}` );

	var total = 0;
	for ( i in [ 0, 1, 2 ] ) {
		total += sorted[i][1];
	}
	console.log( `PART2: ${total}` );

} );
